lib/CustomFields/pages/EditRecordCustomFields/EditRecordCustomFields.js

```diff
@@ -0,0 +1,265 @@
+  changeReduxFormField: PropTypes.func,
+  columnCount: PropTypes.number,
+  entityType: PropTypes.string.isRequired,
+  expanded: PropTypes.bool.isRequired,
+  fieldComponent: PropTypes.func.isRequired,
+  finalFormCustomFieldsValues: PropTypes.objectOf(PropTypes.string),
+  formName: PropTypes.string,
+  isReduxForm: PropTypes.bool,
+  okapi: PropTypes.shape({
+    tenant: PropTypes.string.isRequired,
+    token: PropTypes.string.isRequired,
+    url: PropTypes.string.isRequired,
+  }).isRequired,
+  onToggle: PropTypes.func.isRequired,
+  reduxFormCustomFieldsValues: PropTypes.objectOf(PropTypes.string),
+};
+
+const defaultProps = {
+  columnCount: 4,
+  isReduxForm: false,
+};
+
+const EditRecordCustomFields = ({
+  accordionId,
+  onToggle,
+  expanded,
+  okapi,
+  backendModuleId,
+  backendModuleName,
+  entityType,
+  columnCount,
+  isReduxForm,
+  formName,
+  changeFinalFormField,
+  changeReduxFormField,
+  fieldComponent: Field,
+  reduxFormCustomFieldsValues,
+  finalFormCustomFieldsValues,
+}) => {
+  const {
+    customFields,
+    customFieldsLoaded,
+    customFieldsFetchFailed,
+  } = useCustomFieldsFetch(okapi, backendModuleId, entityType);
+
+  const {
+    sectionTitle,
+    sectionTitleLoaded,
+    sectionTitleFetchFailed,
+  } = useSectionTitleFetch(okapi, backendModuleName.toUpperCase());
+
+  const { calloutRef } = useLoadingErrorCallout(customFieldsFetchFailed || sectionTitleFetchFailed);
+  const dataIsLoaded = sectionTitleLoaded && customFieldsLoaded;
+  const customFieldsIsVisible = customFields?.length && customFields?.some(customField => customField.visible);
+  const accordionIsVisible = dataIsLoaded && customFieldsIsVisible;
+  const visibleCustomFields = customFields?.filter(customField => customField.visible);
+  const formattedCustomFields = chunk(visibleCustomFields, columnCount);
+  const columnWidth = rowShapes / columnCount;
+
+  const collectDataOptionsForSelect = (customField) => {
+    return (customField.selectField?.options?.values || []).map(option => ({
+      label: option,
+      value: option,
+    }));
+  };
+
+  useEffect(() => {
+    const initializeFields = () => {
+      customFields.forEach(cf => {
+        const fieldHasOptions = (
+          cf.type === fieldTypes.SELECT ||
+          cf.type === fieldTypes.RADIO_BUTTON_SET
+        );
+
+        if (isReduxForm) {
+          const fieldIsEmpty = !hasIn(reduxFormCustomFieldsValues, cf.refId);
+
+          if (fieldHasOptions && fieldIsEmpty) {
+            const valueToSet = cf.selectField.defaults[0];
+
+            changeReduxFormField(`customFields.${cf.refId}`, valueToSet);
+          }
+        } else {
+          const fieldIsEmpty = !hasIn(finalFormCustomFieldsValues, cf.refId);
+
+          if (fieldHasOptions && fieldIsEmpty) {
+            const valueToSet = cf.selectField.defaults[0];
+
+            changeFinalFormField(`customFields.${cf.refId}`, valueToSet);
+          }
+        }
+      });
+    };
```

Comment:
```suggestion
          const fieldIsEmpty = !hasIn(reduxFormCustomFieldsValues, cf.refId);

          if (fieldHasOptions && fieldIsEmpty) {
            const valueToSet = cf.selectField.defaults[0];

            if (isReduxForm) {
              changeReduxFormField(`customFields.${cf.refId}`, valueToSet);
            } else {
              changeFinalFormField(`customFields.${cf.refId}`, valueToSet);
            }
          }
      });
    };
```

little bit hard coded, you duplicate yourself

Reply:
@Vladyslav-Velytskyi , we have to know if what type of form is used before calling 
`const fieldIsEmpty = !hasIn(reduxFormCustomFieldsValues, cf.refId);
`
or
`
const fieldIsEmpty = !hasIn(finalFormCustomFieldsValues, cf.refId);
`

Reply:
oh, yes, you're right, sorry, i missed that first arguments are different